### PR TITLE
Add User, Project and Group classes

### PR DESCRIPTION
**Pull request by rsiminel**
@mboudet

I verified that most functionalities still work and everything seems to still be displayed correctly. Is there some rigorous way to make sure these heavy modifications have not broken anything?

### CODE REVIEW

**Comment by lecorguille** [comment]
François Gerbes implemented some Selenium tests.
I don't know if there are still available somewhere?

**Comment by mboudet** [comment]
I'm not sure if they were ever on this repo 🤔

File: manager2/src/app/user/user.component.ts [hunk 1]
```diff
@@ -352,13 +335,13 @@ export class UserComponent implements OnInit {
             }
         }
         if (!found) {
-          this.groups.push({name: this.user.group, new: true});
-          this.missing_group = this.user.group;
+          this.groups.push(this.groupService.mapToGroup({ name: this.user.group, new: true }));
+          this.group_exists = false;
```
Comment (mboudet) [inline manager2/src/app/user/user.component.ts:339]: I 'think' the reason I used `this.missing_group = this.user.group;` and then an `if user.group == this.missing_group` was that if you switch the selected group, you want the 'create group' button to go away.

By setting a static true / false variable, the button will always stay even if I select an existing group

**Review by mboudet — COMMENTED** [review]
Just for clarification: I think you are using user.temp as a placeholder for various 'runtime' variables. Is that the case?

As far as I can see, we only use it for the 'authorized' key. Or is there any other variable we store in there?

File: manager2/src/app/user/user.component.ts (hunk 1 above)
Comment (rsiminel) [reply manager2/src/app/user/user.component.ts]: You're right. Modification reverted.

**Comment by rsiminel** [comment]
> Just for clarification: I think you are using user.temp as a placeholder for various 'runtime' variables. Is that the case?
> As far as I can see, we only use it for the 'authorized' key. Or is there any other variable we store in there?

Yes exactly.
We store the 'authorized' boolean for use in `groups.component` and the 'access' boolean for use in `project.component`.

**Review by mboudet — APPROVED** [review]
Ok, a quick test on my end seems fine. Merging.

**Pull request merged**